tests/test_cli/test_add/test_connection.py

```diff
@@ -517,120 +531,36 @@ def test_add_connection_latest_version(self):
+    """Test add skill in mixed mode that fails with local falls back to remote registry."""
 
-    @classmethod
-    def teardown_class(cls):
-        """Tear the test down."""
-        os.chdir(cls.cwd)
-        try:
-            shutil.rmtree(cls.t)
-        except (OSError, IOError):
-            pass
+    COMPONENT_ID = PublicId(
+        "valory",
+        "http_client",
+        "0.23.0",
+        "bafybeihz3tubwado7j3wlivndzzuj3c6fdsp4ra5r3nqixn3ufawzo3wii",
+    )
+    COMPONENT_TYPE = PackageType.CONNECTION
 
 
-@pytest.mark.skip  # need remote registry
-class TestAddConnectionLocalWhenNoLocalRegistryExists:
+class TestAddConnectionLocalWhenNoLocalRegistryExists(
+    BaseTestAddConnectionLocalWhenNoLocalRegistryExists
+):
     """Test that the command 'aea add connection' fails in local mode when the local registry does not exists."""
 
-    @classmethod
-    def setup_class(cls):
-        """Set the test up."""
-        cls.runner = CliRunner()
-        cls.agent_name = "myagent"
-        cls.cwd = os.getcwd()
-        cls.t = tempfile.mkdtemp()
-        cls.connection_id = str(HTTP_CLIENT_PUBLIC_ID)
-        cls.connection_name = "http_client"
+    COMPONENT_ID = HTTP_CLIENT_PUBLIC_ID
+    COMPONENT_TYPE = PackageType.CONNECTION
 
-        os.chdir(cls.t)
-        result = cls.runner.invoke(
-            cli,
-            [*CLI_LOG_OPTION, "init", "--local", "--author", AUTHOR],
-            standalone_mode=False,
-        )
-        assert result.exit_code == 0, result.stdout
 
-        result = cls.runner.invoke(
-            cli,
-            [*CLI_LOG_OPTION, "create", cls.agent_name],
-            standalone_mode=False,
-        )
-        assert result.exit_code == 0, result.stdout
+@pytest.mark.flaky(reruns=MAX_FLAKY_RERUNS)
+@pytest.mark.integration
+class TestAddConnectionMode(BaseTestAddRemoteMode):
+    """Test case for add connection, --remote mode."""
 
-        os.chdir(cls.agent_name)
-        cls.result = cls.runner.invoke(
-            cli,
-            [*CLI_LOG_OPTION, "add", "--local", "connection", cls.connection_id],
-            standalone_mode=False,
-        )
-
-    def test_exit_code_equal_to_1(self):
-        """Test that the exit code is equal to 1."""
-        assert self.result.exit_code == 1, self.result.stdout
-
-    def test_standard_output_mentions_failure(self):
-        """Test standard output contains information on failure."""
-        assert (
-            "Registry path not provided and local registry `packages` not found in current (.) and parent directory."
-            in self.result.exception.message
-        )
-
-    @classmethod
-    def teardown_class(cls):
-        """Tear the test down."""
-        os.chdir(cls.cwd)
-        try:
-            shutil.rmtree(cls.t)
-        except (OSError, IOError):
-            pass
+    COMPONENT_ID = PublicId(
+        "valory",
+        "http_client",
+        "0.23.0",
+        "bafybeihz3tubwado7j3wlivndzzuj3c6fdsp4ra5r3nqixn3ufawzo3wii",
```

Comment:
Don't use hard coded hashes, use logic similar to https://github.com/valory-xyz/open-autonomy/blob/main/tests/test_autonomy/test_cli/test_build_image.py#L42 test to use hashes from the last release so we don't get any flaky tests in case of a packages get's deleted during garbage collection

Reply:
We have a same function on open-aea here https://github.com/valory-xyz/open-aea/blob/b34f20659bda96b7d92f1338c0f20765dc010052/scripts/check_ipfs_hashes_pushed.py#L65

Reply:
done